ds0_updater.py

```diff
@@ -5,13 +5,36 @@
 import logger
 import conf
 
+# keep it super simple for now because we don't really implement --allowed-keys yet.
+def verify_latest_commit():
+    allowed_keys = ds0_utils.allowed_keys()
+
+    # ensure the latest commit is signed and verified
+    while True:
+        for key in allowed_keys:
+            logger.exe('gpg --import ' + key['file'], expectError=True)
+        output = logger.exe('git log --pretty="format:%G?" --show-signature HEAD^..HEAD')
+
+        if "Can't check signature" in output[0]:
+            logger.info('gpg keys cleared on startup. Trying again...')
+            continue
+
+        verified = False
+        for key in allowed_keys:
+            if key['rsa_check'] in output[0] and key['signature_check'] in output[0]:
+                verified = True
+        if not verified:
+            logger.error('Scripts using a non-signed commit. Please ensure commit is valid.')
+            logger.error('    If it was a missed signature, feel free to open a ticket at https://github.com/riptano/ComboAMI.')
+        break
+
 # Update the AMI codebase if it's its first boot
 if not conf.get_config("AMI", "CompletedFirstBoot"):
     # check if a specific commit was requested
     force_commit = ds0_utils.required_commit()
 
     # update the repo
-    logger.exe('git pull')
+    logger.exe('git pull ' + ds0_utils.repository())
```

Comment:
Ah, you're right - I forgot git would default to the branch if none was specified. This is my mistake and wasn't my intention. In that case I would like to add a `--branch` flag to allow specifying both repository and branch, otherwise users can't keep more than one configuration in a repository.